src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -357,47 +369,159 @@ def executeMcpTestSteps(self, config, test_type):
+          # Check and compare interference for FSS entity
+          self.compareIapAndAggregateResults(fss_cochannel_ap_iap_ref_values, fss_cochannel_aggr_interference,'point')
+          self.compareIapAndAggregateResults(fss_blocking_ap_iap_ref_values, fss_blocking_aggr_interference,'point')
+        elif (fss_low_freq >= interference.FSS_TTC_LOW_FREQ_HZ and 
+                 fss_high_freq <= interference.FSS_TTC_HIGH_FREQ_HZ and 
+                 fss_ttc_flag is True):
+          # Call iap reference model for FSS blocking
+          fss_blocking_ap_iap_ref_values = iap.\
+              performIapForFssBlocking(fss_record, sas_uut, sas_th_fad_objects)
+          # Call aggregate interference model for FSS blocking
+          fss_blocking_aggr_interference = aggregate_interference.\
+              calculateAggregateInterferenceForFssBlocking(fss_record, cbsd_list)
+          # Compare the interference values calculated from both models
+          self.compareIapAndAggregateResults(fss_blocking_ap_iap_ref_values,\
+              fss_blocking_aggr_interference, 'point')
+
+    # Calculate and compare the interference value for ESC protection point 
+    if 'escRecords' in protection_entities:
+      for esc_record in protection_entities['escRecords']:
+        # Call iap reference model for esc
+        esc_ap_iap_ref_values = iap.\
+            performIapForEsc(esc_record, sas_uut, sas_th_fad_objects)
+        # Call aggregate interference model for esc 
+        esc_aggr_interference = aggregate_interference.\
+            calculateAggregateInterferenceForEsc(esc_record, cbsd_list)
+        # Compare the interference values calculated from both models
+        self.compareIapAndAggregateResults(esc_ap_iap_ref_values, esc_aggr_interference, 'point')
+      
+  def compareIapAndAggregateResults(self, ap_iap_ref_values, aggr_interference, entity_type):
+    """Verify aggregate interference is less than or equal to ap_iap_ref value calculated 
+    by IAP model plus a delta for each FSS and ESC sensor protection point, and for at
+    least 95% of the protection points of each PPA and GWPZ.
 
-       Returns:
-         Aggregate Interference list
+    Args:
+      ap_iap_ref_values : Aggregate interference output calculated from IAP 
+                             reference model 
+      aggr_interference : Aggregate interference output calculated from Aggregate
+                          Inteference reference model 
+      entity_type : To identify its a point protection or area protection entity 
+
+    Returns:
+       Pass or Fail based on the comparison of the values within range
     """
-    pass
+    iter_count = 0 # Variable to count the number of interference entries
+    match_count = 0 # Variable to count the number of matching interference entries
+    
+    for lat_dict in ap_iap_ref_values:
+      long_dict = ap_iap_ref_values[lat_dict]
+      long_dict_aggr = aggr_interference[lat_dict]
```

Comment:
shall we also assert that the length of these 2 list are the same?

Reply:
added   the check for comparing the length of the lists